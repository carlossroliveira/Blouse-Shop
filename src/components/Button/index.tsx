// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ButtonSC } from './buttonStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonProps } from './types';

export const Button = (props: ButtonProps) => {
  return (
    <ContainerSC>
      <ButtonSC role="button">{props.text}</ButtonSC>
    </ContainerSC>
  );
};
