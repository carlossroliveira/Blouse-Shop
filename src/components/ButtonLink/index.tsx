// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, LinkSC } from './buttonLinkStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonLinkProps } from './types';

export const ButtonLink = (props: ButtonLinkProps) => {
  return (
    <ContainerSC>
      <LinkSC to={`${props.path}`} onClick={props.onClick}>
        {props.text}
      </LinkSC>
    </ContainerSC>
  );
};
