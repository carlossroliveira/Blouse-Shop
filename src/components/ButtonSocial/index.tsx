// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ButtonSC } from './buttonSocialStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { ButtonSocialProps } from './types';

export const ButtonSocial = (props: ButtonSocialProps) => {
  return (
    <ButtonSC role="button">
      <>
        {props.icon}
        {props.text}
      </>
    </ButtonSC>
  );
};
