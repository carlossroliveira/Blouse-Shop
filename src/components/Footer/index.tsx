// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC, ParagraphSC } from './footerStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { IFooterProps } from './types';

export const Footer = (props: IFooterProps) => {
  const currentYear: number = new Date().getFullYear();

  return (
    <ContainerSC>
      <ParagraphSC Color={props.color}>
        &copy; {currentYear} | Carlos Oliveira
      </ParagraphSC>
    </ContainerSC>
  );
};
