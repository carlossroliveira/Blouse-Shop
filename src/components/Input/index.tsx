// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './inputStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------
import { InputProps } from './types';

export const Input = (props: InputProps) => {
  return (
    <ContainerSC>
      <input
        type={props.type}
        value={props.value}
        onChange={props.onChange}
        placeholder={props.placeholder}
      />
    </ContainerSC>
  );
};
