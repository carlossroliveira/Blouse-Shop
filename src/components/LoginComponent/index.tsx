import React, { FormEvent, useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { useAuth } from '../../hooks/useAuth';

export const LoginComponent = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const NAVIGATION = useNavigate();
  const auth = useAuth();

  const handleSubmita = async (event: FormEvent) => {
    event.preventDefault();

    console.log(email);
    console.log(password);
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  };

  /*  async function onFinish(event: any, email: string, password: string) {
    event.preventDefault();
    try {
      await auth.authenticate(email, password);

      NAVIGATION('/profile');
    } catch (err) {
      alert(err);
    }
  } */

  return (
    <>
      <form onSubmit={handleSubmita}>
        <div>
          <input
            type="text"
            placeholder="email"
            value={email}
            onChange={(event) => setEmail(event.target.value)}
          />
        </div>
        <div>
          <input
            type="text"
            placeholder="Senha"
            value={password}
            onChange={(event) => setPassword(event.target.value)}
          />
        </div>
        <button type="submit">Logar</button>
      </form>
    </>
  );
};
