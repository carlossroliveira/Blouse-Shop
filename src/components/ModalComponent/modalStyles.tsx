// -------------------------------------------------
// Packages
// -------------------------------------------------
import styled from 'styled-components';

export const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
    width: '70rem',
  },
};

export const ContainerSC = styled.section`
  display: flex;
`;

export const DivOneSC = styled.div`
  flex: 1 900px;
  background-color: ${({ theme }) => theme.background.secondary};
`;

export const TitleSC = styled.h2`
  padding: 0px;
  font-size: 24px;
  color: ${({ theme }) => theme.color.secondary};
`;

export const DivInfoSC = styled.div`
  display: flex;
  border-bottom: 1px solid ${({ theme }) => theme.color.secondary};
`;

export const DivInfoTwoSC = styled.div`
  display: flex;
`;

export const ParagraphInfoSC = styled.p`
  font-weight: 600;
  line-height: 16px;
  text-align: center;
  flex: 1 1px;
  color: ${({ theme }) => theme.color.secondary};
`;

export const ImgSC = styled.img`
  width: 10rem;
`;

export const MainSC = styled.div`
  display: flex;
`;

export const DivInputSC = styled.input`
  cursor: text;
  padding: 10px;
  border-radius: 10px;
  margin-right: 1rem;
  border-color: #f80032;
`;

export const DivCodeZipButtonSC = styled.button`
  background-color: white;
  color: #f80032;
  box-shadow: rgb(0 0 0 / 20%) 0px 2px 6px;
  border: 2px solid #f80032;
  font-size: 16px;
  padding: 10px;
  border-radius: 10px;

  transition: filter 0.2s;

  &:hover {
    cursor: pointer;
    filter: brightness(0.9);
  }
`;

//===================== DivTwoSC

export const DivTwoSC = styled.div`
  flex: 1 500px;
  padding: 0 1rem;
  background-color: ${({ theme }) => theme.background.secondary};
`;

export const DivContentOneSC = styled.div`
  padding-bottom: 5px;
  border-bottom: 1px solid ${({ theme }) => theme.color.secondary};
`;

export const DivContentSubSC = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const ParagraphSC = styled.p`
  margin: 5px 0;
  color: ${({ theme }) => theme.color.secondary};
`;

export const DivContentTwoSC = styled.div`
  display: flex;
  flex-direction: column;
  align-items: self-end;
  padding-bottom: 5px;
  border-bottom: 1px solid ${({ theme }) => theme.color.secondary};

  p {
    color: #008000;
  }
`;

export const ButtonSC = styled.button`
  margin-top: 20px;
  width: 100%;
  border-radius: 0.5rem;
  padding: 5px;

  background-color: #f80032;
  font-size: 18px;
  border: 2px solid #f80032;
  color: ${({ theme }) => theme.color.primary};
  box-shadow: rgb(0 0 0 / 20%) 0px 2px 6px;
  transition: filter 0.2s;

  &:hover {
    filter: brightness(0.8);
    cursor: pointer;
  }
`;

export const ButtonBuySC = styled.button`
  width: 40rem;
  height: 4rem;
  background-color: ${({ theme }) => theme.color.quinary};

  border: none;
  font-weight: 700;
  border-radius: 5px;
  text-transform: uppercase;

  box-shadow: ${({ theme }) => theme.boxShadow.secondary};
  transition: filter 0.2s;

  &:hover {
    cursor: pointer;
    filter: brightness(0.9);
  }

  @media (max-width: 975px) {
    width: fit-content;
  }
`;
