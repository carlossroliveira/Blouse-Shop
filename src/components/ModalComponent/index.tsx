// -------------------------------------------------
// Packages
// -------------------------------------------------
import React, { useState } from 'react';
import Modal from 'react-modal';
// -------------------------------------------------
// Styles
// -------------------------------------------------
import { ContainerSC } from './modalStyles';
// -------------------------------------------------
// Types
// -------------------------------------------------

/* const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
}; */

export const ModalComponent = () => {
  const [modalIsOpen, setIsOpen] = useState<boolean>(false);

  const openModal = () => setIsOpen(true);

  const closeModal = () => setIsOpen(false);

  return (
    <ContainerSC>
      <button onClick={openModal}>Open Modal</button>

      <Modal
        isOpen={modalIsOpen}
        onRequestClose={closeModal}
        /* style={customStyles} */
        contentLabel="Example Modal"
      >
        aaaaaaaaaaaaaa
      </Modal>
    </ContainerSC>
  );
};
