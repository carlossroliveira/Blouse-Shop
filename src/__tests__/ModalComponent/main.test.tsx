// -------------------------------------------------
// Packages
// -------------------------------------------------
import React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
// -------------------------------------------------
// Components
// -------------------------------------------------
import { ModalComponent } from '../../components/ModalComponent';
// -------------------------------------------------
// Utils
// -------------------------------------------------
import WithThemeComponent from '../../utils/withThemeComponent';

const LoginWithTheme = WithThemeComponent(ModalComponent);

describe('Testing error ModalComponent', () => {
  it('Should render the button correctly', () => {
    render(<LoginWithTheme />);

    expect(
      screen.getByRole('button', {
        name: /comprar agora/i,
      }),
    ).toBeInTheDocument();
  });

  it('component rendering matches snapshot correctly', () => {
    render(<LoginWithTheme />);
    expect(render(<LoginWithTheme />)).toMatchSnapshot();
  });
});
