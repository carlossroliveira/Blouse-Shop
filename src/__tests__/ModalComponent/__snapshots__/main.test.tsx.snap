// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing error ModalComponent component rendering matches snapshot correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  width: 40rem;
  height: 4rem;
  background-color: #8ca315;
  border: none;
  font-weight: 700;
  border-radius: 5px;
  text-transform: uppercase;
  box-shadow: rgb(38,57,77) 0px 20px 30px -10px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c0:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

@media (max-width:975px) {
  .c0 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
  }
}

<body>
    <div>
      <button
        class="c0"
      >
        Comprar agora
      </button>
    </div>
    <div
      class="ReactModalPortal"
    />
    <div>
      <button
        class="c0"
      >
        Comprar agora
      </button>
    </div>
    <div
      class="ReactModalPortal"
    />
  </body>,
  "container": .c0 {
  width: 40rem;
  height: 4rem;
  background-color: #8ca315;
  border: none;
  font-weight: 700;
  border-radius: 5px;
  text-transform: uppercase;
  box-shadow: rgb(38,57,77) 0px 20px 30px -10px;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
}

.c0:hover {
  cursor: pointer;
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

@media (max-width:975px) {
  .c0 {
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
  }
}

<div>
    <button
      class="c0"
    >
      Comprar agora
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
